internal/mode/static/health.go

```diff
@@ -29,10 +39,16 @@ func (h *healthChecker) readyCheck(_ *http.Request) error {
 }
 
 // setAsReady marks the health check as ready.
-func (h *healthChecker) setAsReady() {
+func (h *healthCheckerImpl) setAsReady() {
 	h.lock.Lock()
 	defer h.lock.Unlock()
 
 	h.ready = true
 	h.firstBatchError = nil
+	close(h.readyIfClosedChan)
+}
+
+// GetReadyIfClosedChannel returns a channel which determines if the NGF Pod is ready.
+func (h *healthCheckerImpl) GetReadyIfClosedChannel() <-chan struct{} {
```

Comment:
```suggestion
func (h *healthCheckerImpl) GetReadyCh() <-chan struct{} {
````

Similar to this https://github.com/nginxinc/nginx-gateway-fabric/pull/1497/files#r1475121453
Channels are used commonly to propagate signals so no need to add "IfClosedChannel".
This is somewhat similar to add "IfReturnsTrue" to a function like below:
```
func ReadyIfReturnsTrue() bool { ... }
```